src/common/components/select-repository-list/index.js

```diff
@@ -117,32 +113,10 @@ export class SelectRepositoryListComponent extends Component {
     }
   }
 
-  onPageLinkClick(pageNumber) {
-    this.props.dispatch(fetchUserRepositories(pageNumber));
-  }
-
-  pageSlice(repositoriesIndex, pageLinks) {
-    if (!pageLinks) {
-      return repositoriesIndex;
-    }
-
-    const PAGE_SIZE = 30; // TODO move to config or state
-    const { next, prev } = pageLinks;
-    let page = [];
-
-    if (next) {
-      page = repositoriesIndex.slice((next - 2) * PAGE_SIZE, (next - 1) * PAGE_SIZE);
-    } else if (prev) {
-      page = repositoriesIndex.slice(prev * 30);
-    }
-
-    return page;
-  }
-
   renderRepoList() {
-    const { ids, error, isFetching, isDelayed, pageLinks } = this.props.repositories;
+    const { ids, error, isFetching, isDelayed } = this.props.repositories;
 
-    if (isFetching) {
+    if (isFetching && ids.length === 0) {
```

Comment:
Making sure we only show loading spinner when there are no repositories yet to avoid 'blinking' when subsequent pages are loaded in background.